### ISSUE TITLE
The connection to the server api.crc.testing:6443 was refused - did you specify the right host or port

### ISSUE THREAD

**Issue by svsiva** [code-ready/crc#1212]
### General information

  * OS: Linux
  * Hypervisor: KVM 
  * Did you run `crc setup` before starting it (Yes/No)? Yes

## CRC version
```bash
# Put the output of `crc version`
crc version
crc version: 1.9.0+a68b5e0
OpenShift version: 4.3.10 (embedded in binary)

```
  
## CRC status
```bash
# Put the output of `crc status`
 crc status
ERRO The connection to the server api.crc.testing:6443 was refused - did you specify the right host or port?
 - exit status 1 


```

## CRC config
```bash
# Put the output of `crc config view`

crc config view
- cpus                                  : 8
- memory                                : 16384
- pull-secret-file                      : /home/kumar/Downloads/pull-secret

```

## Host Operating System
```bash
# Put the output of `cat /etc/os-release` in case of Linux
# put the output of `sw_vers` in case of Mac
# Put the output of `systeminfo` in case of Windows
cat /etc/os-release 
NAME="CentOS Linux"
VERSION="8 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="CentOS Linux 8 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:8"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-8"
CENTOS_MANTISBT_PROJECT_VERSION="8"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="8"



```

### Steps to reproduce

1. crc setup
2. crc config set memory 16384
 3. crc config set cpus 8
 4. crc config set pull-secret-file /home/kumar/Downloads/pull-secret 
5.  crc config view
6. crc start
 7. crc oc-env
8. export PATH="/home/kumar/.crc/bin:$PATH"
 9. eval $(crc oc-env)
10.  oc login -u developer -p developer https://api.crc.testing:6443

  1. 
  2. 
  3. 
  4. 

### Expected
connects to the console and authenticate the user

### Actual
oc login -u developer -p developer https://api.crc.testing:6443
error: dial tcp 192.168.130.11:6443: connect: connection refused - verify you have provided the correct host and port and that the server is currently running.

### Logs

aking call to close driver server
(crc) Calling .Close
Successfully made call to close driver server
Making call to close connection to plugin binary
time="2020-05-04T23:15:10+05:30" level=debug msg="Running '/home/kumar/.crc/bin/oc get co -ojson --kubeconfig /home/kumar/.crc/machines/crc/kubeconfig --context admin --cluster crc'"
time="2020-05-04T23:15:10+05:30" level=debug msg="Command failed: exit status 1"
time="2020-05-04T23:15:10+05:30" level=debug msg="stdout: "
time="2020-05-04T23:15:10+05:30" level=debug msg="stderr: The connection to the server api.crc.testing:6443 was refused - did you specify the right host or port?\n"
time="2020-05-04T23:15:10+05:30" level=error msg="The connection to the server api.crc.testing:6443 was refused - did you specify the right host or port?\n - exit status 1"

You can start crc with `crc start --log-level debug` to collect logs.
Please consider posting this on http://gist.github.com/ and post the link in the issue.

**Comment by svsiva** [comment]
log file posted in gist
https://gist.github.com/svsiva/e4dbbd4245adccaa416c05ed91474739

**Comment by praveenkumar** [comment]
@svsiva Can you try following?

```
$ crc delete
$ crc cleanup
$ crc setup
$ crc start --log-level debug
```

**Comment by svsiva** [comment]
Praveen,
Please have a look at the debug log from 
https://gist.github.com/svsiva/7ecc6fe8455713b104ce3b097690e093

**Comment by praveenkumar** [comment]
@svsiva as per shared logs I can see cluster is started without any issue, are you still not able to login to developer or kubeadmin user like in the logs?

```
INFO To access the cluster, first set up your environment by following 'crc oc-env' instructions 
INFO Then you can access it by running 'oc login -u developer -p developer https://api.crc.testing:6443' 
INFO To login as an admin, run 'oc login -u kubeadmin -p HeJWN-ckbCA-Q96Ds-Sj763 https://api.crc.testing:6443' 
```

**Comment by svsiva** [comment]
@praveenkumar Interestingly I am able to login as developer as well kubeadmin per your instruction once. After that I am randomly getting connection refused errors like below
oc login -u developer -p developer https://api.crc.testing:6443
error: dial tcp 192.168.130.11:6443: connect: connection refused - verify you have provided the correct host and port and that the server is currently running.

**Comment by svsiva** [comment]
@praveenkumar I am able to login to the cluster now per your instruction. What I noticed is the OpenShift api servers is randomly drops the connection.

[kumar@localhost .crc]$ oc login -u developer -p developer https://api.crc.testing:6443
error: dial tcp 192.168.130.11:6443: connect: connection refused - verify you have provided the correct host and port and that the server is currently running.

after few minutes, when I tried it worked.

kumar@localhost .crc]$ oc login -u developer -p developer https://api.crc.testing:6443
Login successful.

You don't have any projects. You can try to create a new project, by running

    oc new-project <projectname>

**Comment by svsiva** [comment]
kumar@localhost .crc]$ oc login -u kubeadmin -p HeJWN-ckbCA-Q96Ds-Sj763 https://api.crc.testing:6443
error: dial tcp 192.168.130.11:6443: connect: connection refused - verify you have provided the correct host and port and that the server is currently running.
[kumar@localhost .crc]$ nmap 192.168.130.1 -p 6443
Starting Nmap 7.70 ( https://nmap.org ) at 2020-05-06 02:45 IST
Nmap scan report for 192.168.130.1
Host is up (0.00015s latency).

PORT     STATE  SERVICE
6443/tcp closed sun-sr-https

wondering why I can't access the admin-console. This is happening randomly

**Comment by svsiva** [comment]
kumar@localhost .crc]$ crc status
ERRO The connection to the server api.crc.testing:6443 was refused - did you specify the right host or port?
 - exit status 1 

[kumar@localhost .crc]$ crc ip
192.168.130.11
[kumar@localhost .crc]$ nmap api.crc.testing -p 6443
Starting Nmap 7.70 ( https://nmap.org ) at 2020-05-06 02:47 IST
Nmap scan report for api.crc.testing (192.168.130.11)
Host is up (0.00018s latency).

PORT     STATE  SERVICE
6443/tcp closed sun-sr-https

Nmap done: 1 IP address (1 host up) scanned in 0.05 seconds

Wondering why connecting to the api server works randomly. How can I login and bring the api server?

**Comment by praveenkumar** [comment]
@svsiva Are you running this CentOS-8 host on baremetal or it is a VM on which your are trying to use the CRC ?

**Comment by svsiva** [comment]
@praveenkumar I am using Baremetal (Dell Laptop) running CentOs-8 for running the CRC.

**Comment by svsiva** [comment]
I have upgraded to CRC version and still the same issues.

[kumar@localhost crc]$ crc version
crc version: 1.10.0+9025021
OpenShift version: 4.4.3 (embedded in binary)

[kumar@localhost crc]$ crc status
CRC VM:          Running
OpenShift:       Not Reachable
Disk Usage:      13.86GB of 32.72GB (Inside the CRC VM)
Cache Usage:     11.9GB
Cache Directory: /home/kumar/.crc/cache
It looks like OpenShift cluster is crashing inside the CRC VM

**Comment by praveenkumar** [comment]
@svsiva have you tried to update your CentOS box and see if there are some kernel or libvirt update? if there is can you please do that and also please put the output of `lscpu` from your Dell laptop.

**Comment by svsiva** [comment]
@praveenkumar my system is upto date with Kernel and libvirt.  

kumar@localhost ~]$ lscpu
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              4
On-line CPU(s) list: 0-3
Thread(s) per core:  2
Core(s) per socket:  2
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               58
Model name:          Intel(R) Core(TM) i5-3340M CPU @ 2.70GHz
Stepping:            9
CPU MHz:             3255.890
CPU max MHz:         3400.0000
CPU min MHz:         1200.0000
BogoMIPS:            5382.40
Virtualization:      VT-x
L1d cache:           32K
L1i cache:           32K
L2 cache:            256K
L3 cache:            3072K
NUMA node0 CPU(s):   0-3
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm cpuid_fault epb pti ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid fsgsbase smep erms xsaveopt dtherm ida arat pln pts md_clear flush_l1d

**Comment by praveenkumar** [comment]
So this looks like you are allocating the more CPU to crc then you actually available on your host. In your host it's `4` cpus only and on the config which you share I can see which is not correct.
```
crc config view
- cpus                                  : 8
- memory                                : 16384
- pull-secret-file                      : /home/kumar/Downloads/pull-secret
```

Can you do following?
```
$ crc delete -f 
$ crc cleanup
$ crc setup
$ crc config set cpus 4
$ crc start --log-level debug 
```

Share the logs if it still fails.

**Comment by svsiva** [comment]
@praveenkumar  Followed all your instructions and started cluster. I am still getting the same error of Openshift cluster not started error. Please see the log from the gist.

https://gist.github.com/svsiva/b1390a78d69c39a9b9dc5d0aae59043f

**Comment by praveenkumar** [comment]
@svsiva Thanks, looking at the logs, looks like initially the apiserver start but then it just died, we have to find out why it is happening, We need some more information around it. Can you also provide some journal logs after the failure and attach it here?

```
ssh -i ~/.crc/machines/crc/id_rsa -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null core@api.crc.testing -- journalctl --no-pager  > journal.log
```

**Comment by svsiva** [comment]
@praveenkumar the journal.log file is more than 800MB. when I gziped it is still 80MB. Let me know where I can upload the file

**Comment by praveenkumar** [comment]
800 MB, this doesn't sound right, have you collected the logs just after the crc start or you waited for sometime because if you wait then it going to increase,

```
$ crc delete -f 
$ crc cleanup
$ crc setup
$ crc config set cpus 4
$ crc start --log-level debug 
```
Just after it finish, collect the journalctl logs.

**Comment by svsiva** [comment]
[journal.log.gz](https://github.com/code-ready/crc/files/4597004/journal.log.gz)
@praveenkumar journal log for your reference

**Comment by praveenkumar** [comment]
@svsiva Thank you, can you also run the following command after the start and put the output?

```
$ ssh -i ~/.crc/machines/crc/id_rsa -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null core@api.crc.testing -- systemctl list-units --state=failed --all
```

**Comment by svsiva** [comment]
@praveenkumar 
[kumar@localhost ~]$ ssh -i ~/.crc/machines/crc/id_rsa -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null core@api.crc.testing -- systemctl list-units --state=failed --all
Warning: Permanently added 'api.crc.testing,192.168.130.11' (ECDSA) to the list of known hosts.
0 loaded units listed.
To show all installed unit files use 'systemctl list-unit-files'.


[kumar@localhost ~]$ ssh -i ~/.crc/machines/crc/id_rsa -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null core@api.crc.testing -- systemctl list-unit-files --state=failed --all
Warning: Permanently added 'api.crc.testing,192.168.130.11' (ECDSA) to the list of known hosts.

0 unit files listed.

**Comment by praveenkumar** [comment]
@svsiva that is not what I expected :( issue which I was suspecting is because of some service failure but looks like this is not the one :( As per your comment https://github.com/code-ready/crc/issues/1212#issuecomment-624257933 you said after a few minutes it works, is it still the case?

**Comment by svsiva** [comment]
One in 25 times, the cluster status shows "started" or "degraded" but terminted in few minutes. At this point I would say CRC is not working for me.

**Comment by praveenkumar** [comment]
This is really sad to hear but on our side we have tested crc on multiple systems but not able to reproduce the issue which you are facing right now :( if you have spare machine then try to use it.

**Comment by svsiva** [comment]
Will CRC work on a CentOs on a virtualbox VM?

**Comment by masini** [comment]
I have the same problem on Mac OS X Catalina and CRC version:

CodeReady Containers version: 1.12.0+6710aff
OpenShift version: 4.4.8 (embedded in binary)

Can you help troubleshooting ? The cluster is unusable.

Thanks.

**Comment by praveenkumar** [comment]
@svsiva no it doesn't work with Centos on VB because then you need nested virtualization and it will become very slow. Can you also try to update the crc version and see if that is fixed?

@masini Please create a new issue with issue template if you are still having issue with mac.

**Comment by mrshengzyzy** [comment]
I have the same problem on CentOS 8 and I found a solution, my  problem is virtualization config is error：
```bash
root@master:~# cat /sys/module/kvm_intel/parameters/nested
N
```
So I tried the following command:
```bash
root@master:~# modprobe -r kvm-intel
root@master:~# modprobe kvm-intel nested=1
```
then `crc start --log-level debug` and it works fine!

Hope can help you!

**Comment by Darkest** [comment]
Im having a very similar problem running crc 1.19 on windows machine using hyper-v virtualization. My workflow looks like:

1.  Create crc vm from the ground up using crc start comand after iva cleaned it all up by running crc cleanup
2.  Crc vm works fine for 1-2 days
3.  Then suddenly web ui stops responding and i get the following error printed in my browser
{"error":"server_error","error_description":"The authorization server encountered an unexpected condition that prevented it from fulfilling the request.","state":"55b1e5c4"}
4. I try to restart the crc cluster using crc stop and then crc start but it refuses to go up right after kubelet service is being started with the error  
DEBU RetryAfter timeout after 16 tries
Failed to update cluster pull secret: Temporary error: ssh command error:
command : oc get secret --context admin --cluster crc --kubeconfig /opt/kubeconfig
err     : Process exited with status 1
output  :  (x16)
5. Going inside the VM i see a repeating logs in journalctl - https://gist.github.com/Darkest/9175b2ef8ab5b5f448f7a270a87aab53

The promblem is really annoying...

**Comment by micspu** [comment]
> 

Same issue I'm facing.any resolution identified?
